common/src/main/java/com/microsoft/identity/common/internal/cache/ActiveBrokerCache.kt

```diff
@@ -60,14 +62,20 @@ class ActiveBrokerCache
         private const val BROKER_METADATA_CACHE_STORE_ON_SDK_SIDE_STORAGE_NAME = "BROKER_METADATA_CACHE_STORE_ON_SDK_SIDE"
 
         /**
-         * The Lock for all [ActiveBrokerCache] instances used by the broker code.
+         * The Mutex for all [ActiveBrokerCache] instances used by the broker code.
+         * (As of May 24, 2023... Kotlin has yet to officially support ReadWriteMutex.
+         *  I don't think it's worth implementing our own (for now).
+         *  If we eventually are seeing a perf hit, sure...)
          **/
-        private val sBrokerSideLock = ReentrantReadWriteLock()
+        private val sBrokerSideLock = Mutex()
```

Comment:
where is this used? same with sSdkSideLock
I believe we are using the provided Mutex

Reply:
Ok but if one is in the Sdkprocess and the other is in the auth process, Each process is in a separate address space. 
So one lock will be ok, they will never collide, right? because they exist in different processes

 this is more for readability and precautions, or there is something I am missing

Reply:
technically yes... but nothing really guarantees that the sdk storage will not be used in broker process (and vice versa)

having separate lock for each storage space is safer.